src/Mangata.ts

```diff
@@ -0,0 +1,265 @@
+/* eslint-disable no-console */
+import { ApiPromise, WsProvider } from '@polkadot/api'
+import { GenericEvent } from '@polkadot/types'
+import { KeyringPair } from '@polkadot/keyring/types'
+import BN from 'bn.js'
+
+import { options } from './utils/options'
+import { RPC } from './services/Rpc'
+import { TX } from './services/Tx'
+import { Query } from './services/Query'
+import { txOptions } from './types'
+
+/**
+ * The Mangata class defines the `getInstance` method that lets clients access the unique singleton instance. Design pattern Singleton Promise is used.
+ */
+export class Mangata {
+  private static instance: Mangata
+  private api: ApiPromise | null
+  private uri: string
+
+  /**
+   * The Mangata's constructor is private to prevent direct
+   * construction calls with the `new` operator.
+   */
+  private constructor(uri: string) {
+    this.api = null
+    this.uri = uri
+  }
+
+  /**
+   * Initialised via isReady & new with specific provider
+   */
+  private async connect() {
+    if (!this.api) {
+      const provider = new WsProvider(this.uri)
+      this.api = await ApiPromise.create(options({ provider }))
+    }
+
+    return this.api
+  }
+
+  /**
+   * The static method that controls the access to the Mangata instance.
+   */
+  public static getInstance(uri: string): Mangata {
+    if (!Mangata.instance) {
+      Mangata.instance = new Mangata(uri)
+    }
+
+    return Mangata.instance
+  }
+
+  /**
+   * Retrieve the underlying API
+   */
+  public async getApi(): Promise<ApiPromise> {
+    return await this.connect()
+  }
+
+  /**
+   * Retrieve the chain name
+   */
+
+  public async getChain(): Promise<string> {
+    const api = await this.connect()
+    return RPC.getChain(api)
+  }
+
+  /**
+   * Retrieve the node name
+   */
+
+  public async getNodeName(): Promise<string> {
+    const api = await this.connect()
+    return RPC.getNodeName(api)
+  }
+
+  /**
+   * Retrieve the node version
+   */
+
+  public async getNodeVersion(): Promise<string> {
+    const api = await this.connect()
+    return RPC.getNodeVersion(api)
+  }
+
+  /**
+   * Retrieve the current nonce
+   */
+
+  public async getNonce(address: string): Promise<BN> {
+    const api = await this.connect()
+    return Query.getNonce(api, address)
+  }
+
+  /**
+   * Disconnect
+   */
+
+  public async disconnect(): Promise<void> {
+    const api = await this.connect()
+    api.disconnect()
+  }
+
+  /**
+   * Create a pool
+   */
+
+  public async createPool(
+    keyRingPair: KeyringPair,
```

Comment:
A minor styling issue: Keyring is a single word, so the right camel is `keyringPair`